### PR TITLE
add isfinite check for all_close

### PR DESCRIPTION
**Pull request by fengleitian**
issue: 
`all_close` return `true` when compare a number or `NaN` with `NaN`.

solution:
1. add `NaN` check.
2. disable/fix tests.

8 test failed after change:
```
[----------] Global test environment tear-down
[==========] 2585 tests from 57 test cases ran. (295590 ms total)
[  PASSED  ] 2577 tests.
[  FAILED  ] 8 tests, listed below:
[  FAILED  ] GPU.backwards_power
[  FAILED  ] CPU.backwards_power
[  FAILED  ] INTERPRETER.backwards_power
[  FAILED  ] cpu_fusion.rnn_fusion_inter_vs_cpu_1lstm_cell
[  FAILED  ] cpu_fusion.rnn_fusion_inter_vs_cpu_1rnn_layer_3lstm_cell
[  FAILED  ] cpu_fusion.rnn_fusion_inter_vs_cpu_2rnn_layer_3lstm_cell
[  FAILED  ] cpu_fusion.fuse_rnn_across_2layer_1timestep
[  FAILED  ] gpu_fusion.rnn_fusion_inter_vs_gpu_1rnn_layer_3lstm_cell
```
Some tests passed before now looks like this:
```
5189 [ RUN      ] cpu_fusion.rnn_fusion_inter_vs_cpu_1lstm_cell
5190 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 52    -nan is not close to 0.964028 at index 0
5191 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 52    -nan is not close to 0.964028 at index 1
5192 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 52    -nan is not close to 0.964028 at index 2
5193 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 52    -nan is not close to 0.964028 at index 3
5194 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 52    -nan is not close to 0.964028 at index 4
5195 [INFO] 2018-11-08T19:19:38z test/./util/all_close.hpp 60    diff count: 1000 out of 1000
5196 /localdisk/tfl/workspace/private-ngraph-cpp/test/cpu_fusion.cpp:2239: Failure
5197 Value of: test::all_close(cpu_results.at(i), int_results.at(i), 1.0e-4f, 1.0e-4f)
5198   Actual: false
5199 Expected: true
```
```
1957 [ RUN      ] CPU.backwards_power
1958 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 52    -nan is not close to -nan at index 0
1959 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 52    -nan is not close to -nan at index 1
1960 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 52    -nan is not close to -nan at index 2
1961 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 52    -nan is not close to -nan at index 3
1962 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 52    -nan is not close to -nan at index 4
1963 [INFO] 2018-11-08T19:19:10z test/./util/all_close.hpp 60    diff count: 36 out of 36
1964 unknown file: Failure
1965 C++ exception with description "Derivatives mismatch between cache and non-cache bprop functions" thrown in the test body.
1966 [  FAILED  ] CPU.backwards_power (11 ms)```
```

### CODE REVIEW

File: test/util/all_close_f.cpp [hunk 1]
```diff
@@ -67,15 +67,22 @@ bool test::all_close_f(const vector<float>& a,
...
     {
         bool is_close_f = close_f(a[i], b[i], mantissa_bits, tolerance_bits);
-        if (!is_close_f)
+        if (!is_close_f && count < 5)
         {
-            NGRAPH_INFO << a[i] << " is not close to " << b[i];
+            NGRAPH_INFO << a[i] << " is not close to " << b[i] << " at index " << i;
             rc = false;
         }
+        count++;
```
Comment (gcwenger) [inline test/util/all_close_f.cpp]: You shouldn't always be incrementing count here.

Comment (fengleitian) [reply test/util/all_close_f.cpp]: right, nice found.

Comment (fengleitian) [reply test/util/all_close_f.cpp]: fixed.

**Comment by rkimballn1** [comment]
In order to merge this we are going to need to either
1) disable the failing tests
or
2) fix the failing tests

File: test/util/all_close.hpp [hunk 2]
```diff
@@ -41,14 +41,24 @@ namespace ngraph
...
+                    {
+                        NGRAPH_INFO << a[i] << " is not close to " << b[i] << " at index " << i;
+                    }
+                    count++;
                     rc = false;
                 }
             }
+            if (!rc)
+            {
+                NGRAPH_INFO << "diff count: " << count << " out of " << a.size();
```
Comment (jbobba) [inline test/util/all_close.hpp]: Would be good to inform that we are only displaying the first 5 mismatches.

Comment (fengleitian) [reply test/util/all_close.hpp]: It will output "xx out of xx has difference"

**Pull request merged**